airflow/www/views.py

```diff
@@ -3026,6 +3026,14 @@ def tree_data(self):
         # avoid spaces to reduce payload size
         return htmlsafe_json_dumps(tree_data, separators=(',', ':'))
 
+    def warn_deployment_exposure(self):
+        deployment_warning = "Deployment is exposed to public internet"
```

Comment:
Could this message maybe be extended with some useful information?

Maybe something like:

```
Recent requests to /robots.txt indicate that this deployment may be accessible to the public internet.

This warning can be disabled by setting webserver.warn_deployment_exposure=False in the configuration.
```

We could also link to a page in the docs with more information on webserver security.